### ISSUE TITLE
Evil mode crashes upon starting emacs 28: Error: evil/:catch: Wrong number of arguments: (3 . 4), 2

### ISSUE THREAD

**Issue by pooyataher** [emacs-evil/evil#1752]
# Issue name

- Bug report
  Evil mode crashes in emacs 28.2

# Environment

Emacs version: 28.2
Operating System: Arch Linux 
Evil version: 1.15.0
Evil installation type: compiled the latest version of evil mode from source using `use-package` and `straight`
Graphical/Terminal: graphical (under xwyaland)
Tested in a `make emacs` session (see [CONTRIBUTING.md](https://github.com/emacs-evil/evil/blob/master/CONTRIBUTING.md#bug-reports)): No

# Reproduction steps
- Start Emacs 28
- Srart Evil mode

# Expected behavior
Evil mode takes effect without any errors upon starting emacs 28

# Actual behavior
- This error will show up upon starting emacs 28:
  Error: evil/:catch: Wrong number of arguments: (3 . 4), 2
- Evil mode does not work at all.  Emacs original keybindings are in place
- by running emacs in debug mode
  `$ emacs --debug-init`
  the following error in emacs is given:
  ```
  Debugger entered--Lisp error: (wrong-number-of-arguments (3 . 4) 2)
  #f(compiled-function (obsolete-name current-name when &optional docstring) "Set OBSOLETE-NAME's function definition to CURRENT-NAME and mark it obsolete.\n\n(define-obsolete-function-alias \\='old-fun \\='new-fun \"28.1\" \"old-fun's doc.\")\n\nis equivalent to the following two lines of code:\n\n(defalias \\='old-fun \\='new-fun \"old-fun's doc.\")\n(make-obsolete \\='old-fun \\='new-fun \"28.1\")\n\nWHEN should be a string indicating when the function was first\nmade obsolete, for example a date or a release number.\n\nSee the docstrings of `defalias' and `make-obsolete' for more details." #<bytecode 0x15d3da019c0917e2>)('evil-ex-shell-command-completion-at-point 'comint-completion-at-point)
  (define-obsolete-function-alias 'evil-ex-shell-command-completion-at-point 'comint-completion-at-point)
  eval-buffer(#<buffer  *load*-795890> nil "/home/pooya/.emacs.d/straight/build/evil/evil-ex.e..." nil t)  ; Reading at buffer position 22912
  load-with-code-conversion("/home/pooya/.emacs.d/straight/build/evil/evil-ex.e..." "/home/pooya/.emacs.d/straight/build/evil/evil-ex.e..." nil t)
  require(evil-ex)
  eval-buffer(#<buffer  *load*-545804> nil "/home/pooya/.emacs.d/straight/build/evil/evil-sear..." nil t)  ; Reading at buffer position 974
  load-with-code-conversion("/home/pooya/.emacs.d/straight/build/evil/evil-sear..." "/home/pooya/.emacs.d/straight/build/evil/evil-sear..." nil t)
  require(evil-search)
  eval-buffer(#<buffer  *load*-125995> nil "/home/pooya/.emacs.d/straight/build/evil/evil.el" nil t)  ; Reading at buffer position 5321
  load-with-code-conversion("/home/pooya/.emacs.d/straight/build/evil/evil.el" "/home/pooya/.emacs.d/straight/build/evil/evil.el" t t)
  require(evil nil t)
  (not (require 'evil nil t))
  (if (not (require 'evil nil t)) (display-warning 'use-package (format "Cannot load %s" 'evil) :error) (condition-case err (progn (evil-mode 1) t) ((debug error) (funcall use-package--warning0 :config err))))
  (condition-case err (if (not (require 'evil nil t)) (display-warning 'use-package (format "Cannot load %s" 'evil) :error) (condition-case err (progn (evil-mode 1) t) ((debug error) (funcall use-package--warning0 :config err)))) ((debug error) (funcall use-package--warning0 :catch err)))
  eval-buffer(#<buffer  *load*-476952> nil "/home/pooya/.emacs.d/custom.el" nil t)  ; Reading at buffer position 662
  load-with-code-conversion("/home/pooya/.emacs.d/custom.el" "/home/pooya/.emacs.d/custom.el" nil nil)
  load("~/.emacs.d/custom.el")
  eval-buffer(#<buffer  *load*> nil "/home/pooya/.emacs.d/init.el" nil t)  ; Reading at buffer position 30
  load-with-code-conversion("/home/pooya/.emacs.d/init.el" "/home/pooya/.emacs.d/init.el" t t)
  load("/home/pooya/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0xc581731e2c1cae2>) #f(compiled-function () #<bytecode -0x1f3c686ddc0cdc35>) t)
  command-line()
  normal-top-level()
  ```

# Further notes
Apparently the fourth optional argument in the signature of the function
`define-obsolete-function-alias`
is now required in the new version 28.2 of emacs.
The function is located in the file
`byte-run.el.gz`
 The function can also be found in the file
`evil-ex.el`

**Comment by axelf4** [comment]
Should have been fixed by #1401. Try updating Evil.

> Apparently the fourth optional argument in the signature of the function `define-obsolete-function-alias` is now required in the new version 28.2 of emacs.

The error `(wrong-number-of-arguments (3 . 4) 2)` is saying two arguments were given when 3 to 4 were required.

**Comment by pooyataher** [comment]
@axelf4 Thanks!  In the Evil source code that `straight` had installed in my emacs, the function `define-obsolete-function-alias` was indeed called with 2 arguments.
I uninstalled `emacs` and installed `emacs-nativecomp` in Arch.  That seems to have fixed the problem!
